ledger/eval.go

```diff
@@ -1525,8 +1525,108 @@ func (vb ValidatedBlock) WithSeed(s committee.Seed) ValidatedBlock {
 	}
 }
 
-// GetBlockAddresses returns all addresses referenced in `block`.
-func GetBlockAddresses(block *bookkeeping.Block) map[basics.Address]struct{} {
+// FoundAddress is a wrapper for an address and a boolean.
+type FoundAddress struct {
+	Address basics.Address
+	Exists  bool
+}
+
+// A ledger interface that Indexer implements. This is a simplified version of the
+// ledgerForEvaluator interface. Certain functions that the evaluator doesn't use
+// in the trusting mode are excluded, and the present functions only request data
+// at the latest round.
+type indexerLedgerForEval interface {
+	BlockHdr(basics.Round) (bookkeeping.BlockHeader, error)
```

Comment:
when do you need different block header rounds ?

Reply:
We only need to previous block header. Do you think I should simplify it?

Reply:
Yes; Naming it LatestBlockHdr() would be a good name... and in `indexerLedgerConnector. BlockHdr` you should test that round == blockHdr.Round(), or error. ( i.e. verify that the ledger wasn't really referring to a header which isn't the latest )

Reply:
btw - same applied for Totals()

Reply:
done